scanpipe/models.py

```diff
@@ -3190,15 +3207,7 @@ def purl(self):
 
     @classmethod
     def extract_purl_data(cls, package_data):
-        purl_data = {}
-
-        for field_name in PURL_FIELDS:
-            value = package_data.get(field_name)
-            if field_name == "qualifiers":
-                value = normalize_qualifiers(value, encode=True)
-            purl_data[field_name] = value or ""
-
-        return purl_data
+        return normalize_package_url_data(package_data)
```

Comment:
Do we need to keep the extract_purl_data since you refactored into an external function?

Reply:
We are using `DiscoveredDependency.extract_purl_data()` in 2 places, and also `DiscoveredPackage.extract_purl_data()` in 2 places. The `DiscoveredPackage` method is exactly `models.normalize_package_url_data()` but the `DiscoveredDependency` method has some extra modifications, so I was looking to keep the same method for purl related fields.